.github: Refactor GHA linux binary builds

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70957
* __->__ #68388

Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

seemethere committed:
.github: Modernize binary build scripts

Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
This PR depends on https://github.com/pytorch/builder/pull/904

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Modernize binary build scripts"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
<details>

<summary> ✨ It finally builds ✨ </summary>

![evil laugh](https://media.giphy.com/media/3o72FfM5HJydzafgUE/giphy.gif)

</details>

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Enable linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

janeyx99 commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -84,6 +90,8 @@ class CIFlowConfig:
     trigger_actor: str = 'pytorchbot'
     root_job_condition: str = ''
     label_conditions: str = ''
+    # Certain jobs might not want to be part of the ciflow/[all,trunk] workflow
```
Shouldn't every job be a part of ciflow/all? I think I'd expect "all" to mean everything

seemethere replied:
I think in the case of binary builds we don't necessarily want them as part of `ciflow/all`. Since they are the most expensive type of workflow we can potentially run

janeyx99 replied:
I agree with that, I'm more concerned that we're adding potential confusion where
ciflow/all -> trunk + scheduled but NOT binaries.

Ideally, people would just use ciflow/trunk instead of ciflow/all most of the time, but this is currently not the case as people have not moved to using ciflow/trunk. 

Random idea that popped into mind: what if we remove the ciflow/all trigger entirely and just have ciflow/trunk, ciflow/scheduled, and ciflow/binaries? This'd get rid of that potential confusion but people who are used to ciflow/all would be confused for a hot minute....

For now I agree we shouldn't add it to ciflow/all

suo replied:
https://fb.quip.com/RcLZAOb9k0qI is relevant here. I think it will let us avoid having to decide which labels map to which jobs. It's just: there is is a label for every workflow. You want that workflow to run, you add that label. No nesting, overlap, or extra semantics.

janeyx99 commented on .github/templates/linux_binary_build_workflow.yml.j2:
```diff
@@ -0,0 +1,251 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: noop
+        run: echo "This job is here so we don't have a bunch of skipped binary builds :D"
```
Assuming this won't be necessary once @suo's push trigger change is pushed out?

seemethere replied:
Yup this'll also have to be changed once we move to that

janeyx99 reviewed (approved):
🎉 looks good overall! Had a few comments

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).

seemethere committed:
Update on ".github: Refactor GHA linux binary builds"


Updates the gpu architectures as well as adding a trigger for
on_pull_request for the binary build workflows so that we can iterate on
this later

TODO:
* Create follow up PR to enable nightly linux GHA builds / disable CircleCI nighlty linux builds

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D33462294](https://our.internmc.facebook.com/intern/diff/D33462294)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33462294).
